journalbeat/input/input.go

```diff
@@ -191,6 +205,26 @@ func (i *Input) publishAll() {
 	}
 }
 
+// isErrSuppressed checks if the error is due to a corrupt journal. If yes, only the first error message
+// is displayed and then it is suppressed for 5 seconds.
+func (i *Input) isErrSuppressed(ctx context.Context, err error, suppressed *atomic.Bool) bool {
+	if strings.Contains(err.Error(), syscall.EBADMSG.Error()) {
+		if suppressed.Load() {
+			return true
+		}
+
+		suppressed.Store(true)
+		go func(ctx context.Context, suppressed *atomic.Bool) {
+			if err := timed.Wait(ctx, 5*time.Second); err == nil {
+				suppressed.Store(false)
+			}
+
+		}(ctx, suppressed)
```

Comment:
shouldn't we return true here?

Reply:
Why? The signature of the goroutine is `func(ctx context.Context, suppressed *atomic.Bool)`.

Reply:
I was rather wondering if we should return the new state of `suppressed`. But I see why `false` should be returned, otherwise we wouldn't have any log.